File: backend/src/models/Invite.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IInvite extends Document {
  code: string;
  serverId: mongoose.Types.ObjectId;
  createdBy: mongoose.Types.ObjectId;
  expiresAt?: Date | null;
  createdAt: Date;
  updatedAt: Date;
}

const InviteSchema = new Schema<IInvite>(
  {
    code: {
      type: String,
      required: true,
      unique: true,
      index: true,
    },
    serverId: {
      type: Schema.Types.ObjectId,
      ref: 'Server',
      required: true,
      index: true,
    },
    createdBy: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    expiresAt: {
      type: Date,
      default: null,
    },
  },
  {
    timestamps: true,
  }
);

export const Invite = mongoose.model<IInvite>('Invite', InviteSchema);

File: backend/app/models/invite.py
from typing import Optional
from datetime import datetime
from pydantic import Field
from beanie import Document
from beanie import PydanticObjectId as ObjectId
from pymongo import IndexModel, ASCENDING

class Invite(Document):
    code: str
    serverId: ObjectId
    createdBy: ObjectId
    expiresAt: Optional[datetime] = None
    createdAt: datetime = Field(default_factory=datetime.utcnow)
    updatedAt: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "invites"
        indexes = [
            IndexModel([("code", ASCENDING)], unique=True),
            IndexModel([("serverId", ASCENDING)]),
        ]
